/** (C) Copyright 2013, Applied Physical Sciences Corp., A General Dynamics Company
 **
 ** Gravity is free software; you can redistribute it and/or modify
 ** it under the terms of the GNU Lesser General Public License as published by
 ** the Free Software Foundation; either version 3 of the License, or
 ** (at your option) any later version.
 **
 ** This program is distributed in the hope that it will be useful,
 ** but WITHOUT ANY WARRANTY; without even the implied warranty of
 ** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 ** GNU Lesser General Public License for more details.
 **
 ** You should have received a copy of the GNU Lesser General Public
 ** License along with this program;
 ** If not, see <http://www.gnu.org/licenses/>.
 **
 */

syntax = "proto2";
option optimize_for = SPEED;
option java_outer_classname = "ServiceDirectoryResponseContainer";
option java_package = "com.aphysci.gravity.protobuf";

package gravity;

message ServiceDirectoryResponsePB
{
	required string id = 1;

	enum ReturnCodes
	{
		SUCCESS = 0;
		DUPLICATE_REGISTRATION = -1;
		REGISTRATION_CONFLICT = -2;
		NOT_REGISTERED = -3;
	}
	optional ReturnCodes returnCode = 2 [default = SUCCESS];
}
